### PR TITLE
AUC / Rules

### PR DESCRIPTION
**Pull request by martinmozina**
##### Issue
1. Got wrong AUC values in domains with binary class
2. Two small changes to rules.py

##### Description of changes
See commit messages. 

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

File: Orange/tests/test_rules.py [hunk 1]
```diff
@@ -36,12 +36,11 @@ def test_base_RuleLearner(self):
         raised).
         """
         base_rule_learner = _RuleLearner()
-        self.assertRaises(NotImplementedError, base_rule_learner.fit,
-                          self.iris.X, self.iris.Y)
+        #self.assertRaises(NotImplementedError, base_rule_learner.fit,
```
Comment (astaric) [inline Orange/tests/test_rules.py]: If the code will not be used any more, just remove these lines. git remembers what was here :)

File: Orange/classification/rules.py [hunk 2]
```diff
@@ -930,8 +929,8 @@ def __init__(self, preprocessors=None, base_rules=None):
         self.cover_and_remove = self.exclusive_cover_and_remove
         self.rule_stopping = self.lrs_significance_rule_stopping
 
-    def fit(self, X, Y, W=None):
-        raise NotImplementedError
+    #def fit_storage(self, data):
```
Comment (astaric) [inline Orange/classification/rules.py]: If this code is not supposed to be here, just remove it

File: Orange/evaluation/scoring.py [hunk 3]
```diff
@@ -186,7 +186,10 @@ def compute_score(self, results, target=None):
         if n_classes < 2:
             raise ValueError("Class variable has less than two values")
         elif n_classes == 2:
-            return self.from_predicted(results, skl_metrics.roc_auc_score)
+            return np.fromiter(
```
Comment (astaric) [inline Orange/evaluation/scoring.py]: A test fail on travis, probably caused by this change. You can see the error returned by test here:

https://travis-ci.org/biolab/orange3/jobs/175538053#L4091

File: Orange/tests/test_rules.py (hunk 1 above)
Comment (martinmozina) [inline Orange/tests/test_rules.py]: Fixed.

File: Orange/classification/rules.py (hunk 2 above)
Comment (martinmozina) [inline Orange/classification/rules.py]: Fixed.

File: Orange/evaluation/scoring.py (hunk 3 above)
Comment (martinmozina) [inline Orange/evaluation/scoring.py]: Fixed.

**Comment by kernc** [comment]
The branch has conflicts. Can you rebase on the latest master?

**Comment by martinmozina** [comment]
Done. Never did that before, hope it is ok.

**Comment by janezd** [comment]
@martinmozina, drop by our lab and somebody will help you removing the stray commits and resolve the conflict.

File: Orange/tests/test_evaluation_scoring.py [hunk 4]
```diff
@@ -65,7 +65,7 @@ def test_call(self):
 
     def test_bayes(self):
         x = np.random.randint(2, size=(100, 5))
-        col = np.random.randint(5)
+       col = np.random.randint(5)
```
Comment (kernc) [inline Orange/tests/test_evaluation_scoring.py]: This change errors.

Comment (martinmozina) [inline Orange/tests/test_evaluation_scoring.py]: I fixed it, but nothing happened. I think the problem is that pull request is still from martinmozina:master branch, but should be from fix-auc branch. At least that's what we tried to achieve with @astaric.

**Comment by janezd** [comment]
Continued in #1826.

**Pull request closed without merging**